HeterogeneousCore/SonicTriton/src/TritonData.cc

```diff
@@ -116,14 +130,16 @@ void TritonInputData::toServer(std::shared_ptr<TritonInput<DT>> ptr) {
   //shape must be specified for variable dims or if batch size changes
   data_->SetShape(fullShape_);
 
-  if (byteSize_ != sizeof(DT))
-    throw cms::Exception("TritonDataError") << name_ << " input(): inconsistent byte size " << sizeof(DT)
+  std::unique_ptr<TritonConverterBase<DT>> converter = createConverter<DT>();
```

Comment:
actually, this could also simplify the implementation of `createConverter()` somewhat. it wouldn't need to have a return type, because it could just store the converter in the `std::any` member variable directly.

Reply:
so I thought a bit about this and I think we run into the issue of needing to know the data type at a point when the code is currently not aware of it. This is because `createConverter()` (or really the actual factory `get()` inside it) needs to know the type of factory. So we have to call `createConverter()` inside `toServer()` or `fromServer()`, and this seems like it defeats the purpose of using `std::any` here. Please correct me if Im wrong

Reply:
The point is to avoid actually re-constructing the converter every time. Just make it once, then store it as a class member and reuse it. `std::any` allows storing as a class member.

Reply:
Right, but I want to do this in the constructor presumably, which doesn't yet know the data type, so I can't call createConverter since it's templated. Or is the suggestion here to play some game to only create the converter the first time `toServer()` is called?

Reply:
Yes: check if the `any` is null when cast to the correct type, and then create it, otherwise use the existing one.

Reply:
Ah, ok, this makes sense to me now, thanks. In order to do this I'll need to make `fromServer` a nonconst function, since it needs to be able to create the converter. I want to make sure you agree with this change before I go ahead

Reply:
Ah, that does break the interface a bit. Let's try making the member variable mutable instead; that should be acceptable here.

Reply:
ok, sounds good